Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -83,14 +125,18 @@ extension ResourceOptions {
             return false
         }
 
-        return
-            (accessToken == other.accessToken) &&
-            (baseURL == other.baseURL) &&
-            (cachePath == other.cachePath) &&
-            (assetPath == other.assetPath) &&
-            (tileStorePath == other.tileStorePath) &&
-            (loadTilePacksFromNetwork == other.loadTilePacksFromNetwork) &&
-            (cacheSize == other.cacheSize)
+        return (accessToken == other.accessToken)
+            && ((baseURL == other.baseURL)
+                    || (baseURL == nil)
+                    || (other.baseURL == nil))
```

Comment:
why should they be considered equal if one has a nil baseURL and the other does not?

Reply:
This is due to a mismatch - when looking at fetched ResourceOptions, these can be non-nil (showing the default, which is what `nil` meant in the first place). It's certainly not ideal.

Reply:
Do we have any non-test code that depends on this implementation?